library/psa_crypto_slot_management.h

```diff
@@ -93,6 +122,7 @@ psa_status_t psa_internal_allocate_key_slot( psa_key_handle_t *handle,
 psa_status_t psa_validate_persistent_key_parameters(
     psa_key_lifetime_t lifetime,
     psa_key_file_id_t id,
+    psa_se_drv_table_entry_t **p_drv,
```

Comment:
Should this be `const psa_se_drv_table_entry_t **p_drv`? I don't see a reason we need to change the driver table it self in this function, although we do want to change the value of the pointer and don't want to make the pointer const.

Reply:
I don't think so: it would require the caller to pass a `const psa_se_drv_table_entry_t **`, and then the caller('s caller) wouldn't be able to change any part of the table entry, specifically the mutable part of the context that the driver is allowed to modify (`driver->context->transient_data`).

I did try to make more things const around driver table entries, but I didn't find a way.

Reply:
Implicitly casting to a function that accepts a const is fine. It doesn't have to be actually const for the caller('s caller).

Reply:
What do you mean by “implicitly casting”? I can't find a way to add the `const`s that I'd like to add without adding (explicit) casts, which are a maintenance burden at best and cause undefined behavior due to lying to the compiler at worst.

Reply:
I played around with this as well. Looks I was wrong about this. We can't avoid explicit casts in C if we want const correctness here, due to the nested pointer.